packages/@aws-cdk/aws-lambda/lib/event-source-filter.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * Filter rules for Lambda event filtering
+ */
+export class FilterRule {
```

Comment:
This is an excellent start here. I'd like to see the return be a class on its own so that the contract for `filters` can be `{ [key: string]: FilterRule }`. [Schedule](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-events/lib/schedule.ts) is an excellent example of what I'd like to see here.

Reply:
Cool, this approach is much better

Reply:
This is good, I believe I can use this approach

Reply:
If you feel like this can be further refined, I'm all for it, but given your note above, I'm not entirely sure it's feasible. Let me know if you want to take a crack at it still or if you just want to re-review with what's already done. Let me know what you think.

Reply:
You're right, FilterRule's returns are so diverse that it's difficult to create a single contract

Reply:
@marciocadev quick question as I was a bit confused at first when trying to use this myself. I read the docs on https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.FilterRule.html but that does not mention the wrapping with `lambda.FilterCriteria.filter`. Without this it seems the filter isn't actually applied, but with it everything works as expected.
Is this simply an omission in the docs or should it work without wrapping it?

Reply:
I am in the same situation as @Lewenhaupt - also spent a while in the docs before I found this PR and realized I had to use the `FilterCriteria` to make it work.
The incorrect syntax is also mentioned here: 
https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda_event_sources.DynamoEventSource.html

Reply:
@Spjak I just spent an hour crying that it wasnt working :) 

Thankyou!